src/libraries/DelegateTokenReverts.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.0;
+
+import {DelegateTokenErrors as Errors} from "src/libraries/DelegateTokenErrors.sol";
+import {DelegateTokenConstants as Constants} from "src/libraries/DelegateTokenConstants.sol";
+import {DelegateTokenStructs as Structs} from "src/libraries/DelegateTokenStructs.sol";
+import {IERC721Receiver} from "openzeppelin/token/ERC721/IERC721Receiver.sol";
+
+library DelegateTokenReverts {
+    function onInvalidERC721ReceiverCallback(address from, address to, uint256 delegateTokenId, bytes calldata data) internal {
+        if (to.code.length != 0 && IERC721Receiver(to).onERC721Received(msg.sender, from, delegateTokenId, data) != IERC721Receiver.onERC721Received.selector) {
+            revert Errors.NotERC721Receiver(to);
+        }
+    }
+
+    function onInvalidERC721ReceiverCallback(address from, address to, uint256 delegateTokenId) internal {
+        if (to.code.length != 0 && IERC721Receiver(to).onERC721Received(msg.sender, from, delegateTokenId, "") != IERC721Receiver.onERC721Received.selector) {
+            revert Errors.NotERC721Receiver(to);
+        }
+    }
```

Comment:
The 721 callback is needed for the `DelegateToken` `safeTransferFrom` functions, otherwise it would be an incorrect implementation of the 721 standard. You can just call `transferFrom` on `DelegateToken` if you don't want to support these callbacks.